candi/bashible/common-steps/node-group/089_setup_timer_to_run_docker_stuck_containers_cleaner.sh.tpl

```diff
@@ -0,0 +1,153 @@
+# Copyright 2021 Flant JSC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+{{- if eq .cri "Docker" }}
+bb-event-on 'docker-stuck-containers-cleaner-changed' '_on_docker_stuck_containers_cleaner_changed'
+_on_docker_stuck_containers_cleaner_changed() {
+{{ if ne .runType "ImageBuilding" }}
+  systemctl daemon-reload
+  systemctl restart docker-stuck-containers-cleaner.timer
+  systemctl restart docker-stuck-containers-cleaner
+{{ end }}
+  systemctl enable docker-stuck-containers-cleaner.timer
+  systemctl enable docker-stuck-containers-cleaner
+}
+
+bb-sync-file /var/lib/bashible/docker-stuck-containers-cleaner.sh - << "EOF"
+#!/bin/bash
+
+image_pattern="^(dev-)?registry.deckhouse.io"
+failed_container_kills_times=5
+log_period_minutes=20
+
+# $@ - messages
+function echo_err() {
+  echo "$@" 1>&2;
+}
+
+function get_stuck_containers() {
+  sed_pattern='s/^.+Container ([a-z0-9]{64}) failed to exit within [0-9]+ seconds of signal 15 - using the force.+$/\1/p'
+  journalctl --since "$log_period_minutes min ago" -u docker.service | \
+    sed -nr "$sed_pattern" | \
+    sort | \
+    uniq -c | \
+    awk '{ if (int($1)>='"$failed_container_kills_times"') print $2 }'
+}
+
+function parse_arguments() {
+  while [[ $# -gt 0 ]]; do
+      key="$1"
+      case $key in
+        -p|--image-pattern)
+          image_pattern="$2"
+          shift # past argument
+          shift # past value
+          ;;
+        -c|--failed-kills-count-in-log)
+          failed_container_kills_times="$2"
+          shift # past argument
+          shift # past value
+          ;;
+        -t|--log-period-minutes)
+          log_period_minutes="$2"
+          shift # past argument
+          shift # past value
+          ;;
+      esac
+    done
+}
+
+function main() {
+  parse_arguments "$@"
+
+  echo "---"
+  echo "Start check docker containers are in stuck. "
+  echo "Use '$image_pattern' for match container image."
+  echo "A container will be considered stuck if it is mentioned $failed_container_kills_times times in dockerd log"
+  echo "Logs will be gotten for last $log_period_minutes minutes"
+  echo "---"
```

Comment:
I think this ouput is not needed. It's garbage output.

Reply:
ok. I'll remove it